### PR TITLE
fix: sdk connection issues

### PR DESCRIPTION
**Pull request by abretonc7s**
**Description**

- Fixes session error on when reconnecting on existing session (ios).
- Also fixed indirect issue linked to connection stability.

### CODE REVIEW

File: app/components/Views/SDKSessionsManager/SDKSessionItem.tsx [hunk 1]
```diff
@@ -67,11 +67,27 @@ export const SDKSessionItem = ({
...
+      _sessionName = connection.originatorInfo.url;
+    }
+
+    if (
+      connection.originatorInfo?.icon &&
+      connection.originatorInfo?.icon?.length > 0
+    ) {
+      setIcon(connection.originatorInfo?.icon);
+    }
+
```
Comment (omridan159) [inline app/components/Views/SDKSessionsManager/SDKSessionItem.tsx:90]:
```suggestion
    let _sessionName = connection.id;

    const { title, url, icon } = connection.originatorInfo || {};

    if (title?.length > 0) {
      _sessionName = title;
    } else if (url?.length > 0) {
      _sessionName = url;
    }

    // Update the icon state if icon information is available
    if (icon?.length > 0) {
      setIcon(icon);
    }
```

Comment (omridan159) [reply app/components/Views/SDKSessionsManager/SDKSessionItem.tsx:90]: More clean  way

File: app/core/SDKConnect/SDKConnect.ts [hunk 2]
```diff
@@ -413,10 +424,16 @@ export class Connection extends EventEmitter2 {
...
-            await wait(1000);
+          if (!this.receivedDisconnect) {
+            while (!this.isReady) {
+              await wait(1000);
+            }
+            this.sendAuthorized();
+          } else {
+            // Reset state to continue communication after reconnection.
+            this.isReady = true;
+            this.receivedDisconnect = false;
```
Comment (omridan159) [inline app/core/SDKConnect/SDKConnect.ts:435]:
```suggestion
  if (!this.receivedDisconnect) {
    let attempts = 0;

    while (!this.isReady && attempts < 10) {
      await wait(1000);
      attempts++;
    }

    this.sendAuthorized();
  } else {
    // Reset state to continue communication after reconnection.
    this.isReady = true;
    this.receivedDisconnect = false;
  }
```

Are we sure that we don't want to create a max attempts logic?

LMk WDYT? :)

File: app/components/Views/SDKSessionsManager/SDKSessionItem.tsx (hunk 1 above)
Comment (abretonc7s) [reply app/components/Views/SDKSessionsManager/SDKSessionItem.tsx:90]: this would also gives linting ts errors:
eg:'icon.length' is possibly 'undefined'.

**Review by omridan159 — DISMISSED** [review]

**Review by elefantel — DISMISSED** [review]
Looks good.

**Review by elefantel — APPROVED** [review]
Looks good

**Comment by christopherferreira9** [comment]
**Context:**
in the flow Connect Dapp -> Approve connection in MMM -> Kill MMM -> Tap sign, the user was being prompt with a connect request instead of the sining prompt. This made that the user would need to approve the connection he did before making it be misleading into making the user think it is a new connection. 
The second issue with this is that if the user taps cancel instead of connect in this resuming prompt the connection would be deleted and the Dapp would never know that so the user gets into a stale state.
The third issue occurs with the Signing feedback mentioned [here](https://github.com/MetaMask/metamask-mobile/issues/6994) that was indirectly caused by this reconnection flow making the appearance of this toaster forever hidden.

**Error Video:**

https://github.com/MetaMask/metamask-mobile/assets/104831203/6c296f77-f9a2-42e4-ab91-b03ca9ef9b84



**Acceptance Criteria:**
- After killing the MetaMask Wallet you're able to resume with sign (meaning that the user is able to tap sign on Dapp and get deeplinked to MetaMask)
- After the MetaMask Wallet resumes the user is prompt with the sign approval
- Approving the sign request presents the success message in MetaMask
- Approving the sign request presents the rejection message in MetaMask


**SDKs and Dapps tested:**
- Android SDK
- Unity SDK
- iOS SDK
- Javascript SDK: Create-React-App; NextJS; ReactNative; ElectronJS; NodeJS

**Video Fix for sign on resume:**
Uploading ios_fix.mov…

**Video fix for the signing feedback:**

https://github.com/MetaMask/metamask-mobile/assets/104831203/0fb2f804-b96a-4b94-bc54-304cd7d05b2d

**Pull request merged**